Api/Services/OrderService.cs

```diff
@@ -1,8 +1,119 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Reservant.Api.Data;
+using Reservant.Api.Models;
+using Reservant.Api.Models.Dtos.Order;
+using Reservant.Api.Validation;
+using System.Reflection.Metadata.Ecma335;
+using FluentValidation.Results;
+using Reservant.Api.Validators;
+using Reservant.Api.Models.Dtos.OrderItem;
+using Reservant.Api.Models.Dtos.MenuItem;
+using System.Linq;
+
 namespace Reservant.Api.Services;
 
 /// <summary>
 /// Service for managing orders
 /// </summary>
-public class OrderService
+public class OrderService(ApiDbContext context, ValidationService validationService)
 {
+    /// <summary>
+    /// Update status of the order by updating the list of states of included menu items
+    /// </summary>
+    /// <param name="id">order id</param>
+    /// <param name="request">request containing list of updated menu items and employees that work on them</param>
+    /// <param name="user"></param>
+    /// <returns></returns>
+    public async Task<Result<bool>> UpdateOrderStatusAsync(int id, UpdateOrderStatusRequest request, User user)
+    {
+        var order = await context.Orders
+            .Where(o => o.Id == id)
+            .Include(o => o.Employees)
+            .Include(o => o.OrderItems)
+            .Include(o => o.Visit)
+            .ThenInclude(v => v.Restaurant)
+            .ThenInclude(r => r.Group)
+            .FirstOrDefaultAsync();
+        //does order with id exist?
+        if (order is null)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = null,
+                ErrorCode = ErrorCodes.NotFound,
+                ErrorMessage = "Order not found"
+            };
+        }
+        if (order.Visit.Restaurant is null)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = null,
+                ErrorCode = ErrorCodes.NotFound,
+                ErrorMessage = $"Restaurant not found, visit: visitID - {order.VisitId}, restaurantID - {order.Visit.Restaurant?.Id}"
+            };
+        }
+
+        //is it in the correct restaurant group?
+        if (order.Visit.Restaurant.Group.OwnerId != user.EmployerId)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = null,
+                ErrorCode = ErrorCodes.AccessDenied,
+                ErrorMessage = ErrorCodes.AccessDenied
+            };
+        }
+
+        //update status of the order's menuitems
+
+        foreach (UpdateOrderItemStatusRequest item in request.Items)
+        {
+            order.OrderItems.First(oi => oi.MenuItemId == item.MenuItemId).Status = item.Status;
```

Comment:
to nie łapie błędów. jeśli nie ma menu itema o podanym id, dostajemy InvalidOperationException